tributors/main/orcid.py

```diff
@@ -132,77 +130,119 @@ def get_orcid_token():
+
+    print("\n\n%s\n======================================================" % email)
+    for idx, r in enumerate(results):
 
+        # Limit is ten results, count starting at 0
+        idx = idx + 1
+        if idx > 10:
+            break
+
+        record = "  Name: %s, %s\n  Orcid: %s" % (
+            r["family-names"],
+            r["given-names"],
+            r["orcid-id"],
+        )
+        if r["institution-name"]:
+            record = "%s\n  Institutions: %s" % (
+                record,
+                ", ".join(r["institution-name"]),
+            )
+        if r["other-name"]:
+            record = "%s\n  Other Names: %s" % (record, ", ".join(r["other-name"]))
+        if r["email"]:
+            record = "%s\n  Email: %s" % (record, r["email"])
+        if not interactive:
+            print("%s\n" % record)
+        else:
+            print("[%s]\n%s\n" % (idx, record))
+
+    # If interactive, ask for choice prompt
+    if interactive:
+        choices = [str(i) for i, _ in enumerate(results)] + ["s", "S", "skip"]
+        prefix = "1:%s or s to skip" % ("10" if len(results) > 10 else len(results))
+        choice = choice_prompt(
+            "Please enter a choice, or s to skip.",
+            choices=choices,
+            choice_prefix=prefix,
+            multiple=False,
+        )
 
-def get_orcid(email, token, name=None):
-    """Given a GitHub repository address, retrieve a lookup of contributors
-       from the API endpoint. We look to use the GITHUB_TOKEN if exported
-       to the environment, and exit if the response has any issue
+        if not choice or choice in ["s", "S", "skip"]:
+            return
+
+        # Return the orcid identifier
+        return results[int(choice) - 1]["orcid-id"]
+
+
+def get_orcid(email, name=None, interactive=False):
+    """Get an orcid identifier for a given email or name.
     """
-    # We must have a token, and an email OR name
-    if not token and (not email and not name):
+    # We must have an email OR name
+    if not email and not name:
         return
 
     # First look for records based on email
     orcid_id = None
     if email:
-        url = "https://pub.orcid.org/v3.0/search/?q=email:%s" % email
-        orcid_id = record_search(url, token, email)
+        url = "https://pub.orcid.org/v3.0/expanded-search?q=email:%s" % email
+        orcid_id = record_search(url, email, interactive)
 
     # Attempt # 2 will use the first and last name
     if name is not None and not orcid_id:
 
-        # If no comma, likely a space delimiter
         delim = "," if "," in name else " "
         cleaner = "," if delim == " " else " "
 
         parts = name.split(delim)
+
+        # No go if only a first or last name
+        if len(parts) == 1:
+            bot.debug(f"Skipping {name}, first and last are required for search.")
+            return orcid_id
+
         last, first = parts[0].strip(cleaner), " ".join(parts[1:]).strip(cleaner)
```

Comment:
as I have mentioned in the email there is a special meaning for `","` in the `name`.  It is either "First Names LastName"  (thus not really allowing to tell First Names from the Last one when it is composite like in Matteo Visconti de Ollegio Castello") or "Last Name, First Names" (so "Visconti di Oleggio Castello, Matteo", ORCID https://orcid.org/0000-0001-7931-5272 -- may be use as a test?). So I think code would benefit from making it explicit even if query does not even care any longer (since might start to in the future).  Code could explicitly check for `,` and use .split or .rsplit, e.g.

```ipython
# no comma, "incorrect" example but just for consistency:
In [10]: _ = first, last = map(str.strip, "Matteo Visconti di Oleggio Castello".rsplit(None, 1))
In [11]: _
Out[11]: ['Matteo Visconti di Oleggio', 'Castello']

# there is comma -- split by it into last, first
In [12]: _ = last, first = map(str.strip, "Visconti di Oleggio Castello, Matteo".split(',', 1))
In [13]: _
Out[13]: ['Visconti di Oleggio Castello', 'Matteo']
```

Reply:
Sure, can you show me explicitly what you want? I don't understand why you are piping the results both into first, last and an empty variable `_`.

Reply:
`_` was just to show values right away... forget about it -- I just demonstrated use of `split` and `rsplit` to split one way or another

Reply:
more explicit:
```python
if ',' in name:
   last, first = map(str.strip, name.split(',', 1))
else:
   first, last = map(str.strip, name.rsplit(None, 1))
```

Reply:
I don't think this approach is cleaner / better because we'd still have to wrap it in some try/catch to account for running into single word names (GitHub logins) that should be skipped, and are nicely handled now:

```
        # No go if only a first or last name
        if len(parts) == 1:
            bot.debug(f"Skipping {name}, first and last are required for search.")
            return orcid_id
```

Reply:
My main point that first isn't necessarily first, and last isn't necessarily last in current code... Confusing/incorrect

Reply:
The naming doesn’t matter though because we search for each without an attribute (e.g given-names).

Reply:
from the original comment: " So I think code would benefit from making it explicit even if query does not even care any longer (since might start to in the future)."